### PR TITLE
data/selinux: permit init_t to remount snappy_snap_t

### PR DESCRIPTION
**Pull request by bboozzoo**
Systemd services that get a new mount namespace on startup fail due to SELinux
blocking system from remounting /var/lib/snapd/snap into the new mount ns.

Since we cannot declare fs_type() and generate contexts through genfscon outside
of core policy, add an explicit permission for init_t.

Corresponding denial:
```
type=AVC msg=audit(1559625610.991:67): avc:  denied  { remount } for  pid=578
         comm="(resolved)"
         scontext=system_u:system_r:init_t:s0
         tcontext=system_u:object_r:snappy_snap_t:s0
         tclass=filesystem permissive=0
```
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1708991

cc @Conan-Kudo

### CODE REVIEW

File: tests/regression/rhbz-1708991/task.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        distro_install_package systemd-resolved
+    fi
+
+    getenforce > enforcing.mode
+
+    # Enable enforcing mode, our policy is already marked as permissive, so we
+    # will get audit entries but the program will not be stopped by SELinux
+    setenforce 1
+    ausearch --checkpoint stamp -m AVC || true
+
```
Comment (stolowski) [inline tests/regression/rhbz-1708991/task.yaml]: Would it make sense to have a debug: section and print some extra selinux-related diagnostics in case we have any failures in the future? (FWTW I looked at existing selinux spread tests and we don't do that atm.)

File: tests/regression/rhbz-1708991/task.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+restore: |
+    setenforce "$(cat enforcing.mode)"
+    rm -f stamp enforcing.mode
+
+execute: |
+    # prepare does systemctl daemon-reexec to workaround:
```
Comment (mvo5) [inline tests/regression/rhbz-1708991/task.yaml]: I don't see any `systemctl daemon-reexec` in this test. Maybe rephrase the comment to something like "the global prepare does ..." and also outline why this comment is relevant for this test?

File: tests/regression/rhbz-1708991/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/regression/rhbz-1708991/task.yaml]: We already show the audit log in spread.yaml.

File: tests/regression/rhbz-1708991/task.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    if [[ "$SPREAD_SYSTEM" == centos-7-* ]]; then
+        distro_install_package systemd-resolved
+    fi
+
+    getenforce > enforcing.mode
+
+    # Enable enforcing mode, our policy is already marked as permissive, so we
+    # will get audit entries but the program will not be stopped by SELinux
+    setenforce 1
+    ausearch --checkpoint stamp -m AVC || true
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply tests/regression/rhbz-1708991/task.yaml]: Yes. The command fails when checkpoint does not exist.

**Comment by Conan-Kudo** [comment]
@zyga @bboozzoo If you know how to fix it, propose a fix here: https://github.com/fedora-selinux/selinux-policy

PRs are welcome there. :)
